blockchain/hash.go

```diff
@@ -26,3 +55,43 @@ type Marshaller interface {
 func HashSum(m Marshaller) Hash {
 	return sha256.Sum256(m.Marshal())
 }
+
+// CompareTo compares two hashes, it returns true if the operation of the first hash on the second hash specified by the comparator is true, and false otherwise
+func CompareTo(h1 Hash, h2 Hash, comparator int) bool {
```

Comment:
math.big uses the exact same arbitrary constants for comparison, so if we wanted to compare two hashes we would have to convert to math.big and then use the same constants